From 674c23856fdadadd8453e16cb0b48ef1d1bed0d5 Mon Sep 17 00:00:00 2001
From: "Yan, Like" <like.yan@intel.com>
Date: Tue, 11 Jun 2019 13:00:42 +0800
Subject: [PATCH 1418/1418] ACRN: don't init SCI/GL handler for ACRN RT VM

Because there is no IOAPIC/PIC on ACRN RT VM by design, SCI/GL handler init
will fail and then disable the ACPI, which eventually results in the RT VM
power off failure. And for ACRN RT VM, only power off initiated internal
RT OS is supported, so it won't break function to not init SCI/GL handler.

This commit skip the SCI/GL handler initilization if CONFIG_ACRN and
CONFIG_PREEMPT_RT_BASE defined.

Signed-off-by: Yan, Like <like.yan@intel.com>
---
 drivers/acpi/bus.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/acpi/bus.c b/drivers/acpi/bus.c
index d2e29a1..e1af02e 100644
--- a/drivers/acpi/bus.c
+++ b/drivers/acpi/bus.c
@@ -1154,7 +1154,11 @@ static int __init acpi_bus_init(void)
 		}
 	}
 
+#if defined(CONFIG_ACRN) && defined(CONFIG_PREEMPT_RT_BASE)
+	status = acpi_enable_subsystem(ACPI_NO_ACPI_ENABLE | ACPI_NO_HANDLER_INIT);
+#else
 	status = acpi_enable_subsystem(ACPI_NO_ACPI_ENABLE);
+#endif
 	if (ACPI_FAILURE(status)) {
 		printk(KERN_ERR PREFIX
 		       "Unable to start the ACPI Interpreter\n");
-- 
2.7.4

